Alt bias variance plot

eric-pedersen opened the pull request:
Hi folks, 

This is the last major change the reviewers requested: extending the bias-variance plot to include multiple noise levels and multiple simulations (to illustrate the amount of variation in fitting these models). 

This was the original plot:

![capture](https://user-images.githubusercontent.com/1554249/50792905-878aef80-1294-11e9-9e2a-3d9b6f2546b3.PNG)


And here's what the new plot with updated changes looks like: 

![capture](https://user-images.githubusercontent.com/1554249/50792877-6c1fe480-1294-11e9-8950-67cde834cf42.PNG)

(details on the text changes are in the last couple commits on this branch).

Thoughts on the changes? Worth it to go with the new figure?

Cheers,

Eric

dill commented:
This is looking good to me Eric. Looking throught the compiled pdf at this point in this branch, I think the text holds together and the graphic looks good. A couple of thoughts though:

1. the black line in the (a) part of the figure is a little thick, and obscures the other lines
2. can you make this bigger? Maybe making the y axis label for (b) smaller or getting it to fit on one line (suggest the x/y labels are "fitted" and "true" then explain the rest in the caption?)
3. not sure about the "<=" at the bottom -- how much < was the smallest?

I think that's all for now, thanks for taking the time to put this together! I think it makes the point really well!

eric-pedersen commented:
Thanks for the feedback! 

> 1. the black line in the (a) part of the figure is a little thick, and obscures the other lines

Definitely; I had made it bigger here as there were more lines, but I agree, right now it is too large. 

> 2. can you make this bigger? Maybe making the y axis label for (b) smaller or getting it to fit on one line (suggest the x/y labels are "fitted" and "true" then explain the rest in the caption?)

That should be doable; I'll try it now.

> 3. not sure about the "<=" at the bottom -- how much < was the smallest?

The smallest was ~ 10e-10, which is quite small. Really, the issue is when mgcv is trying to fit a straight line, which would have a wiggliness of zero. So on a log-log plot, it's trying to go to negative infinity. I think the "less than" option is really the only way to do this in log-log space, but I'm definitely open to suggestions on how to make this clearer.

dill commented:
Okay, I think that all sounds fine then! I'm happy with the "<=" if you can add some chat to the caption?

eric-pedersen commented:
> Okay, I think that all sounds fine then! I'm happy with the "<=" if you can add some chat to the caption?

Alright, I've gone through and adjusted. Here's the new caption:

> Illustration of bias that can arise from assuming equal smoothness for all group levels (model *S*, red lines) versus allowing for intergroup variation in smoothness (model *I*, blue lines) across a range of signal-to-noise ratios, holding the group-level signals constant. The true function for each group level is shown in black. b) Distribution of wiggliness (as measured by the integral of the squared 2nd derivative)  of the estimated function for each replicate for each group level for model *S* (red) and model *I* (blue), versus the true wiggliness of the function for that grouping level, with the black line indicating the one-to-one line. Points below (above) the black line indicate that a given model estimated the curve as less (more) wiggly than the true curve used to generate the data. Estimated wiggliness less than $10^{-3}$ was truncated for visual clarity, as **mgcv** estimated effectively straight lines for several groups, corresponding to a wiggliness of 0, which would not appear on a log-scaled plot.


And the new plot:

![capture](https://user-images.githubusercontent.com/1554249/50839378-2c0d4000-132e-11e9-8a3b-29a782c6aff6.PNG)

I also switched to a dot histogram for the derivative plot, and truncated it at 1e-3 instead of 1e-6, to save a bit more vertical space, and to give more room for panel a. Does this work for you?

dill commented:
This looks good to me!

eric-pedersen commented:
Great! I'll merge it in to the main branch (I've already merged in the other PRs).

gavinsimpson commented:
Figured it was easiest to add this here: @eric-pedersen could you move the legend on panel b to the top rather than have it inside the figure? It took me a little while to notice this.

That way it matches all the other panel (and other figures in terms of legends outside figures). Will need to adjust the relative heights of the rows in the plot a little to keep the current ratio, which I think is good.

gavinsimpson commented:
Another suggestion; do we need signal / noise as a fraction in the facet labels? We'd save some space if we used "SNR = xx" and explain SNR in the the figure caption.
